browser/static/js/canvas.js

```diff
@@ -206,15 +191,31 @@ class CanvasView {
       this.sWidth = newWidth;
     }
     if (oldZoom !== newZoom) {
-      const scaledWidth = this.scale * this.width;
-      const scaledHeight = this.scale * this.height;
-      const propX = canvasPosX / scaledWidth;
-      const propY = canvasPosY / scaledHeight;
+      const propX = canvasPosX / this.scaledWidth;
+      const propY = canvasPosY / this.scaledHeight;
       const dx = propX * (newWidth - oldWidth);
       const dy = propY * (newHeight - oldHeight);
       this.pan(dx, dy);
     }
   }
 
+  drawImage(ctx, image, padding = 0) {
+    ctx.clearRect(padding, padding, this.width, this.height);
+    ctx.drawImage(
+      image,
+      this.sx, this.sy,
+      this.sWidth, this.sHeight,
+      padding, padding,
+      this.scaledWidth,
+      this.scaledHeight
+    );
+  }
 
-}
\ No newline at end of file
+  updateCursorPosition(x, y, padding) {
+    this.updatePos(x, y, padding, this);
+  }
```

Comment:
could we just rename `updatePos` to `updateCursorPosition` and get rid of this method?

Reply:
yes! good catch.